lib/iaas/AzureClient.js

```diff
@@ -0,0 +1,192 @@
+    return this.storage
+      .deleteBlobAsync(container, file)
+      .then(response => logger.info(`Response of blob deletion from cloud storage for ${file} :`, response))
+      .catch(BaseCloudClient.providerErrorTypes.Unauthorized, err => {
+        logger.error(err.message);
+        throw new Unauthorized('Authorization at the cloud storage provider failed');
+      })
+      .catch(BaseCloudClient.providerErrorTypes.Forbidden, err => {
+        logger.error(err.message);
+        throw new Forbidden('Authentication at the cloud storage provider failed');
+      })
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${file}' not found`));
+  }
+
+  download(options) {
+    return utils.streamToPromise(this.storage.createReadStream(
+        options.container, options.remote))
+      .catchThrow(BaseCloudClient.providerErrorTypes.NotFound, new NotFound(`Object '${options.remote}' not found`));
+  }
+
+  upload(options, buffer) {
+    return new Promise((resolve, reject) => {
+      function cleanup() {
+        uploadedStream.removeListener('finish', onfinish);
+        uploadedStream.removeListener('error', onerror);
+      }
+
+      function onerror(err) {
+        cleanup();
+        reject(err);
+      }
+
+      function onfinish() {
+        cleanup();
+        resolve(JSON.parse(buffer));
+      }
+
+      let uploadedStream = this.storage
+        .createWriteStreamToBlockBlob(
+          options.container, options.remote);
+
+      uploadedStream.on('error', onerror);
+      uploadedStream.on('finish', onfinish);
+      uploadedStream.end(buffer);
+    });
+  }
+
+  uploadJson(container, file, data) {
+    if (arguments.length < 3) {
+      data = file;
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .upload({
+        container: this.containerName,
+        remote: file
+      }, new Buffer(JSON.stringify(data, null, 2), 'utf8'));
+  }
+
+  downloadJson(container, file) {
+    if (arguments.length < 2) {
+      file = container;
+      container = this.containerName;
+    }
+    return this
+      .download({
+        container: this.containerName,
+        remote: file
+      })
+      .then((data) => {
+        const response = JSON.parse(data);
+        response.trigger = response.trigger === CONST.BACKUP.TRIGGER.MANUAL ? CONST.BACKUP.TRIGGER.SCHEDULED : response.trigger;
+        //The above conversion is done to handle existing CRON Jobs which set this trigger as 'manual' even for scheduled Jobs
+        //Above conversion can be removed and code changes can be revereted 14 days after the current fix goes live
+        return response;
+      })
+      .catchThrow(SyntaxError, new NotFound(`Object '${file}' not found`));
+  }
+
+  deleteSnapshot(snapshotName) {
+    return Promise
+      .try(() => {
+        return this.computeClient
+          .snapshots
+          .deleteMethodAsync(this.settings.resource_group, snapshotName)
```

Comment:
Basic validation of this.settings in constructor and logging warning/errors will help in identifying the issues early on in case of some misses in setting right props.

Reply:
if the resource group is not present we should return error. Shall i return the error? in that case broker monit process will not start which is required also in my opinion. But thing is all other config params are check as at the beginning the connection clients are created using them. Only this resource group is pending :P

Reply:
Done check for resource_group

Reply:
ok